giles-eco-requests/src/main/java/edu/asu/diging/gilesecosystem/requests/ICompletionNotificationRequest.java

```diff
@@ -46,5 +46,16 @@ public interface ICompletionNotificationRequest extends IRequest {
     void setExtractionDate(String extractionDate);
 
     String getExtractionDate();
-
-}
\ No newline at end of file
+    
+    /**
+    Checks if the the file is a derived file.
+    @return {@code true} if the file is derived from a non base component, {@code false} otherwise.
+    */
+    public abstract boolean isDerivedFile();
```

Comment:
sorry, I know this has been here for a while. Thinking about this again though, I think a better approach is to have a field in the request class that hold the id of the service asking to store a file. The different component can then look at that field and decide if they accept files stored by the component for processing. E.g. Tardis would process files from Cepheus but not from Imogen. I can imagine adding other components that generate derived files that still should be processed by Tardis (or a similar problem with other new components). Does that make sense?